flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -630,8 +631,9 @@ private void setupResultPartitionBookkeeping(
 
 		for (ResultPartitionDeploymentDescriptor resultPartitionDeploymentDescriptor : partitionsRequiringRelease) {
 			final ResultPartitionID resultPartitionId = resultPartitionDeploymentDescriptor.getShuffleDescriptor().getResultPartitionID();
+			final IntermediateDataSetID dataSetId = resultPartitionDeploymentDescriptor.getResultId();
```

Comment:
I would rather put this loop into `filterPartitionsRequiringRelease` which would become:
`Set<ResultPartitionID> filterAndStartTrackingPartitionsRequiringRelease(...)`
and keep `setupResultPartitionBookkeeping` altogether untouched

Reply:
I quite like that the start/stop of the tracking is based on the same collection.